diff --git a/addons/point_of_sale/static/src/app/screens/payment_screen/payment_screen.js b/addons/point_of_sale/static/src/app/screens/payment_screen/payment_screen.js
index 39d3b66cc63..d1e97b0c3d9 100644
--- a/addons/point_of_sale/static/src/app/screens/payment_screen/payment_screen.js
+++ b/addons/point_of_sale/static/src/app/screens/payment_screen/payment_screen.js
@@ -271,9 +271,10 @@ export class PaymentScreen extends Component {
         }
     }
     async _finalizeValidation() {
-        if (this.currentOrder.is_paid_with_cash() || this.currentOrder.get_change()) {
-            this.hardwareProxy.openCashbox();
-        }
+        // Always open cashbox (by default, Odoo only opens cashbox for cash payments)
+        // if (this.currentOrder.is_paid_with_cash() || this.currentOrder.get_change()) {
+        this.hardwareProxy.openCashbox();
+        // }
 
         this.currentOrder.date_order = serializeDateTime(luxon.DateTime.now());
         for (const line of this.paymentLines) {
